### PR TITLE
RATIS-2136 Fixed the issue that RatisServer could not start after frequent restarts

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see RATIS-2136

### CODE REVIEW

**Comment by OneSizeFitsQuorum** [comment]
@szetszwo PTAL!

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLogBase.java [hunk 1]
```diff
@@ -274,7 +274,7 @@ public final void open(long lastIndexInSnapshot, Consumer<LogEntryProto> consume
         e -> commitIndex.updateToMax(e.getMetadataEntry().getCommitIndex(), infoIndexChange));
     state.open();
 
-    final long startIndex = getStartIndex();
+    final long startIndex = Math.max(getStartIndex(), lastIndexInSnapshot + 1);
     if (startIndex > LEAST_VALID_LOG_INDEX) {
       purgeIndex.updateIncreasingly(startIndex - 1, infoIndexChange);
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLogBase.java:279]: I am thinking that `purgeIndex` should not depend on `lastIndexInSnapshot`.  How about changing to `purgeIndex.updateToMax` instead?
```java
      purgeIndex.updateToMax(startIndex - 1, infoIndexChange);
```

**Review by szetszwo — COMMENTED** [review]
@OneSizeFitsQuorum , thanks for working on this!  Please see the comment inlined.

File: ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLogBase.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLogBase.java]: Great idea！Has fixed

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**